src/main/java/com/monumental/controllers/MonumentController.java

```diff
@@ -211,21 +217,53 @@ public MonumentBulkValidationResult validateMonumentCSV(@ModelAttribute BulkCrea
      * @param request - Contains the field mapping and the file to process
-     * @return BulkCreateResult - Object representing the results of the Bulk Monument Create operation
+     * @return AsyncJob - Object containing the Id of the job created and the current value of the Future object
      */
-    @PostMapping(value = "/api/monument/bulk/create")
-    public List<Monument> createMonumentCSV(@ModelAttribute BulkCreateMonumentRequest request) throws IOException {
+    @PostMapping("/api/monument/bulk/create/start")
+    public AsyncJob startBulkCreateMonumentJob(@ModelAttribute BulkCreateMonumentRequest request) throws IOException {
         BulkCreateMonumentRequest.ParseResult parseResult = request.parse(this.monumentService);
 
         MonumentBulkValidationResult validationResult = this.monumentService.validateMonumentCSV(
                 parseResult.csvContents, parseResult.mapping, parseResult.zipFile
         );
 
-        return this.monumentService.bulkCreateMonuments(
+        /* TODO: This is not a particularly easy way of creating AsyncJobs, I can't think of a way to abstract
+         * it away currently because the AsyncJob must be passed to the CompletableFuture method, and the CompletableFuture
+         * must be passed to the AsyncJob, making it difficult to do so dynamically
+         */
+        AsyncJob job = this.asyncMonumentService.createJob();
+        job.setFuture(this.asyncMonumentService.bulkCreateMonuments(
+                job,
                 new ArrayList<CsvMonumentConverterResult>(validationResult.getValidResults().values())
-        );
+        ));
+        return job;
+    }
+
+    /**
+     * Check the progress of a create bulk monuments job
+     * @param id - Id of the job to check
+     * @return AsyncJob - Object containing the Id of the job and the current value of the Future object
+     */
+    @GetMapping("/api/monument/bulk/create/progress/{id}")
+    public AsyncJob getBulkCreateMonumentJob(@PathVariable Integer id) {
+        return this.asyncMonumentService.getJob(id);
+    }
+
+    /**
+     * Get the final result of a create bulk monuments job. If the job is not completed yet this will wait for it to
+     * complete, so be sure to call getBulkCreateMonumentJob and check the status before calling this
+     * @param id - Id of the job to get the result of
+     * @return List<Monument> - The monuments created
+     * @throws ExecutionException - Can be thrown by Java if the future encountered an exception
+     * @throws InterruptedException - Can be thrown by Java if the future encountered an exception
+     */
+    @GetMapping("/api/monument/bulk/create/result/{id}")
+    @SuppressWarnings("unchecked")
+    public List<Monument> getBulkCreateMonumentJobResult(@PathVariable Integer id)
+            throws ExecutionException, InterruptedException {
+        return (List<Monument>) this.asyncMonumentService.getJob(id).getFuture().get();
```

Comment:
This *should* be called after the frontend *knows* that the job is done. Theoretically, you could call it early, in which case Java will wait for the job to finish. I may add a check to this so that Java aborts if it sees that the job isn't done yet, to prevent the thread waiting for the job to finish.

What's returned here is the final result of the operation, in this case the list of created monuments